custom_components/miele/sensor.py

```diff
@@ -540,6 +552,55 @@ def device_state_attributes(self):
         return attributes
 
 
+class MieleEnergyConsumptionSensor(SensorEntity):
+    def __init__(self, hass, device, key):
+        self._hass = hass
+        self._device = device
+        self._key = key
+
+        self._attr_device_class = DEVICE_CLASS_ENERGY
+        self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        self._attr_native_unit_of_measurement = 'kWh'
+
+    @property
+    def device_id(self):
+        """Return the unique ID for this sensor."""
+        return self._device["ident"]["deviceIdentLabel"]["fabNumber"]
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this sensor."""
+        return self.device_id + "_" + self._key
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        ident = self._device["ident"]
+
+        result = ident["deviceName"]
+        if len(result) == 0:
+            return ident["type"]["value_localized"] + " " + _map_key(self._key)
+        else:
+            return result + " " + _map_key(self._key)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        device_state = self._device["state"]
+
+        if "ecoFeedback" in device_state and device_state["ecoFeedback"] is not None:
+            if "currentEnergyConsumption" in device_state["ecoFeedback"]:
+                return device_state["ecoFeedback"]["currentEnergyConsumption"]["value"]
```

Comment:
Ideally we should check that `device_state["ecoFeedback"]["currentEnergyConsumption"]["unit"]` is still `kWh` here, and do a conversion if it isn't. I'll try to get that done.

Reply:
I added a check for the unit and automatic conversion in case of `Wh`.